xbmc/addons/AddonManager.cpp

```diff
@@ -1088,6 +1088,16 @@ bool CAddonMgr::IsFromOfficialRepo(const AddonPtr& addon, bool bCheckAddonPath)
          std::any_of(m_officialAddonRepos.begin(), m_officialAddonRepos.end(), comparator);
 }
 
+AddonOriginType CAddonMgr::GetAddonOriginType(const AddonPtr& addon) const
+{
+  if (m_systemAddons.find(addon->ID()) != m_systemAddons.end())
+    return AddonOriginType::SYSTEM;
+  else if (!addon->Origin().empty())
+    return AddonOriginType::REPOSITORY;
+  else
+    return AddonOriginType::MANUAL;
+}
```

Comment:
returns:
**repository** for `skin.estouchy`, `resource.images.weatherfanart.single`, `service.xbmc.versioncheck` and more
**system** for `repository.xbmc.org` (which is correct, but it's also a repo)
**manual** for a zip-installed priv. repo

for a quick check i modified a label in DialogAddonInfo.xml:
```
<control type="label">
  <width>470</width>
  <height>40</height>
  <label>$INFO[ListItem.Property(addon.origintype),[COLOR button_focus]$LOCALIZE[31150]:[/COLOR] ]</label>
  <visible>!String.IsEmpty(ListItem.AddonOrigin) + !ListItem.Property(addon.isbinary)</visible>
</control>
```

Reply:
Estouchy should come back as system based on the sql output.

Reply:
Fixed now